eventuals/interrupt.h

```diff
@@ -50,56 +49,36 @@ class Interrupt final {
 
     Interrupt* interrupt_ = nullptr;
     Callback<void()> callback_;
-    Handler* next_ = nullptr;
   };
 
   Interrupt()
-    : handler_(this, []() {}) {}
+    : placeholder_handler_(this, []() {}) {}
 
-  bool Install(Handler* handler) {
-    CHECK(handler->next_ == nullptr) << "handler is already installed";
-
-    handler->next_ = head_.load(std::memory_order_relaxed);
-
-    do {
-      // Check if the interrupt has already been triggered.
-      if (handler->next_ == nullptr) {
-        return false;
-      }
-    } while (!head_.compare_exchange_weak(
-        handler->next_,
-        handler,
-        std::memory_order_release,
-        std::memory_order_relaxed));
+  [[nodiscard]] bool Install(Handler* handler) {
+    auto* stored = handler_.load();
+    if (!stored) {
+      // Means that interrupt has already been triggerred.
+      return false;
+    }
 
-    return true;
+    return handler_.compare_exchange_weak(stored, handler);
```

Comment:
I'd like to make this stronger:
```suggestion
    // NOTE: we should be the *only* ones trying to install an interrupt at a time,
    // so the only way we fail to install ourselves is if the interrupt got triggered.
    bool installed = handler_.compare_exchange_weak(
        stored,
        handler,
        std::memory_order_release,
        std::memory_order_relaxed);

  CHECK(installed || stored == nullptr);;

  return installed;
```